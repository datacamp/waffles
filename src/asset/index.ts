export { default as BarChart3D } from './output/bar-chart';
export { default as Bookmark3D } from './output/bookmark';
export { default as Branch3D } from './output/branch';
export { default as Calendar3D } from './output/calendar';
export { default as Certification3D } from './output/certification';
export { default as Checkmark3D } from './output/checkmark';
export { default as Code3D } from './output/code';
export { default as Database3D } from './output/database';
export { default as Direction3D } from './output/direction';
export { default as DoubleArrow3D } from './output/double-arrow';
export { default as Edit3D } from './output/edit';
export { default as Education3D } from './output/education';
export { default as Email3D } from './output/email';
export { default as Finance3D } from './output/finance';
export { default as FourArrows3D } from './output/four-arrows';
export { default as Globe3D } from './output/globe';
export { default as Import3D } from './output/import';
export { default as Laptop3D } from './output/laptop';
export { default as Love3D } from './output/love';
export { default as MagnifyingGlass3D } from './output/magnifying-glass';
export { default as MailboxEmpty3D } from './output/mailbox-empty';
export { default as Mobile3D } from './output/mobile';
export { default as Paper3D } from './output/paper';
export { default as Path3D } from './output/path';
export { default as People3D } from './output/people';
export { default as Rocket3D } from './output/rocket';
export { default as Share3D } from './output/share';
export { default as TargetArrow3D } from './output/target-arrow';
export { default as Target3D } from './output/target';
export { default as Team3D } from './output/team';
export { default as TimerSleep3D } from './output/timer-sleep';
export { default as Timer3D } from './output/timer';
export { default as Track3D } from './output/track';
export { default as Trophy3D } from './output/trophy';
export { default as AvatarPlaceholder } from './output/avatar-placeholder';
