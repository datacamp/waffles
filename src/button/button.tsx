import React from 'react';
import { css } from '@emotion/react';
import { transparentize } from 'polished';

import { tokens } from '../tokens';

const sizeMap = {
  small: {
    sizing: tokens.sizing.small,
    spacing: tokens.spacing.small,
  },
  medium: {
    sizing: tokens.sizing.medium,
    spacing: tokens.spacing.medium,
  },
  large: {
    sizing: tokens.sizing.large,
    spacing: tokens.spacing.medium,
  },
};

const regularVariantMap = {
  primary: {
    color: tokens.colors.navy,
    backgroundColor: tokens.colors.green,
    hoverColor: tokens.colors.greenLight,
    borderColor: 'transparent',
  },
  secondary: {
    color: tokens.colors.navy,
    backgroundColor: 'transparent',
    hoverColor: transparentize(1 - tokens.opacity.low, tokens.colors.navy),
    borderColor: tokens.colors.navy,
  },
  plain: {
    color: tokens.colors.blueDarkText,
    backgroundColor: 'transparent',
    hoverColor: transparentize(1 - tokens.opacity.low, tokens.colors.navy),
    borderColor: 'transparent',
  },
  destructive: {
    color: tokens.colors.navy,
    backgroundColor: tokens.colors.red,
    hoverColor: tokens.colors.redLight,
    borderColor: 'transparent',
  },
};

const invertedVariantMap = {
  ...regularVariantMap,
  secondary: {
    color: tokens.colors.white,
    backgroundColor: 'transparent',
    hoverColor: transparentize(1 - tokens.opacity.low, tokens.colors.white),
    borderColor: transparentize(1 - tokens.opacity.medium, tokens.colors.white),
  },
  plain: {
    color: tokens.colors.blueDark,
    backgroundColor: 'transparent',
    hoverColor: transparentize(1 - tokens.opacity.low, tokens.colors.white),
    borderColor: 'transparent',
  },
};

const buttonStyle = css`
  display: inline-flex;
  align-items: center;
  justify-content: center;
  vertical-align: middle;
  font-family: ${tokens.fontFamilies.sansSerif};
  font-weight: ${tokens.fontWeights.bold};
  font-size: ${tokens.fontSizes.medium};
  line-height: ${tokens.lineHeights.tight};
  margin: 0;
  padding: 0;
  outline: 0;
  border-style: solid;
  border-radius: ${tokens.borderRadius.medium};
  border-width: ${tokens.borderWidth.medium};
  cursor: pointer;
  transition: background-color 125ms ease-out;
`;

type IconProps = {
  size?: 'medium' | 'small' | 'xsmall';
};

type ButtonIconOnlyProps = {
  children?: never;
  icon: React.ComponentType<IconProps>;
  iconLeft?: never;
  iconRight?: never;
  'aria-label': string;
};

type ButtonRegularProps = {
  children: React.ReactNode;
  icon?: never;
  iconLeft?: React.ComponentType<IconProps>;
  iconRight?: React.ComponentType<IconProps>;
  'aria-label'?: string;
};

type ButtonProps = {
  variant?: 'primary' | 'secondary' | 'plain' | 'destructive';
  size?: 'small' | 'medium' | 'large';
  fullWidth?: boolean;
  inverted?: boolean;
} & (ButtonIconOnlyProps | ButtonRegularProps) &
  React.ButtonHTMLAttributes<HTMLButtonElement>;

function Button({
  variant = 'primary',
  size = 'medium',
  fullWidth = false,
  inverted = false,
  icon: Icon,
  iconLeft: IconLeft,
  iconRight: IconRight,
  children,
  ...restProps
}: ButtonProps) {
  const variantMap = inverted ? invertedVariantMap : regularVariantMap;

  return (
    <button
      css={css`
        ${buttonStyle}
        height: ${sizeMap[size].sizing};
        min-width: ${sizeMap[size].sizing};
        width: ${fullWidth ? '100%' : 'auto'};
        padding-left: ${!Icon && sizeMap[size].spacing};
        padding-right: ${!Icon && sizeMap[size].spacing};
        color: ${variantMap[variant].color};
        background-color: ${variantMap[variant].backgroundColor};
        border-color: ${variantMap[variant].borderColor};

        &:hover {
          background-color: ${variantMap[variant].hoverColor};
        }

        &:active {
          background-color: ${variantMap[variant].backgroundColor};
        }
      `}
      {...restProps}
    >
      {Icon && <Icon size="medium" />}
      {IconLeft && <IconLeft size="medium" />}
      {children && (
        <span
          css={css`
            ${IconLeft && `padding-left: ${tokens.spacing.small};`}
            ${IconRight && `padding-right: ${tokens.spacing.small};`}
          `}
        >
          {children}
        </span>
      )}
      {IconRight && <IconRight size="medium" />}
    </button>
  );
}

export default Button;
