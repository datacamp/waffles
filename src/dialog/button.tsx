import { forwardRef } from 'react';

import ButtonInternal from './button-internal';

const Button = forwardRef(ButtonInternal);

export default Button;
