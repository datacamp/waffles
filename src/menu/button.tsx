import React from 'react';

import type { PolymorphicComponentProps } from '../helpers';
import { Button } from '../button';
import { useMenu } from './menu-context';
import { buttonStyle } from './styles';

type MenuButtonBaseProps = {
  variant?: React.ComponentProps<typeof Button>['variant'];
  /* An icon displayed to the left. Could be any [icon](/components/icon) from Waffles (use default `medium` size) or a custom component. */
  iconLeft?: React.ReactNode;
  /* An icon displayed to the right. Could be any [icon](/components/icon) from Waffles (use default `medium` size) or a custom component. */
  iconRight?: React.ReactNode;
  /* The content inside the button. Most of the time should be a plain text. */
  children: React.ReactNode;
  /* [skip docs] */
  inverted?: boolean;
  /* [skip docs] */
  index?: number;
};

type MenuButtonProps<T extends React.ElementType = 'button'> =
  PolymorphicComponentProps<T, MenuButtonBaseProps>;

function MenuButton<T extends React.ElementType = 'button'>({
  index = 0,
  onClick,
  ...restProps
}: MenuButtonProps<T>) {
  const { listRef, setIsOpen, getItemProps, triggerRef } = useMenu();

  function handleClick(event: React.MouseEvent<HTMLButtonElement>) {
    onClick?.(event);
    setIsOpen(false);
    // @ts-expect-error: focus() not recognized
    triggerRef.current?.focus();
  }

  return (
    // @ts-expect-error: spread props out of sync
    <Button
      {...getItemProps({
        onClick: handleClick,
        ...restProps,
      })}
      ref={(node) => (listRef.current[index] = node)}
      role="menuitem"
      onClick={handleClick}
      css={buttonStyle()}
    />
  );
}

export default MenuButton;
