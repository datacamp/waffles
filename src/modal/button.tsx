import React, { forwardRef } from 'react';

import { Button as ButtonBase } from '../button';
import { buttonStyle } from './styles';

type ButtonProps = { autoFocus?: boolean } & React.ComponentProps<
  typeof ButtonBase
>;

function ButtonInternal(
  { autoFocus = false, ...restProps }: ButtonProps,
  ref: React.Ref<HTMLElement>,
) {
  return (
    <ButtonBase
      {...restProps}
      data-autofocus={autoFocus ? true : null}
      ref={ref}
      css={buttonStyle()}
    />
  );
}

const Button = forwardRef(ButtonInternal);

export default Button;
