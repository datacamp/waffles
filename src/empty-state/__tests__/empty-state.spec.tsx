import { render } from '@testing-library/react';

import { EmptyState } from '../index';
import { Group, Matrix, Branch } from '../../icon';
import { Button } from '../../button';
import { Rocket3d } from '../../asset';

jest.mock('../../icon', () => {
  return {
    Group: () => 'Group',
    Branch: () => 'Branch',
    Matrix: () => 'Matrix',
  };
});

jest.mock('../../asset', () => {
  return {
    Rocket3d: () => 'Rocket3d',
  };
});

describe('Dialog', () => {
  it('renders snapshot', async () => {
    const { container } = render(
      <EmptyState title="Title Heading" image={<Rocket3d />}>
        <EmptyState.Content>Lorem ipsum dolor sit amet.</EmptyState.Content>
        <EmptyState.List>
          <EmptyState.ListItem icon={<Group />}>
            Auctor eu augue
          </EmptyState.ListItem>
          <EmptyState.ListItem icon={<Branch />}>
            Tristique senectus et netus
          </EmptyState.ListItem>
          <EmptyState.ListItem icon={<Matrix />}>
            Mus mauris vitae ultricies
          </EmptyState.ListItem>
        </EmptyState.List>
        <EmptyState.Content>
          <Button>Button</Button>
        </EmptyState.Content>
      </EmptyState>,
    );

    expect(container).toMatchSnapshot();
  });

  it('renders snapshot in column direction', async () => {
    const { container } = render(
      <EmptyState title="Title Heading" image={<Rocket3d />} direction="column">
        <EmptyState.Content>Lorem ipsum dolor sit amet.</EmptyState.Content>
        <EmptyState.List>
          <EmptyState.ListItem icon={<Group />}>
            Auctor eu augue
          </EmptyState.ListItem>
          <EmptyState.ListItem icon={<Branch />}>
            Tristique senectus et netus
          </EmptyState.ListItem>
          <EmptyState.ListItem icon={<Matrix />}>
            Mus mauris vitae ultricies
          </EmptyState.ListItem>
        </EmptyState.List>
        <EmptyState.Content>
          <Button>Button</Button>
        </EmptyState.Content>
      </EmptyState>,
    );

    expect(container).toMatchSnapshot();
  });

  it('renders snapshot in inverted', async () => {
    const { container } = render(
      <EmptyState title="Title Heading" image={<Rocket3d />} inverted>
        <EmptyState.Content>Lorem ipsum dolor sit amet.</EmptyState.Content>
        <EmptyState.List>
          <EmptyState.ListItem icon={<Group />}>
            Auctor eu augue
          </EmptyState.ListItem>
          <EmptyState.ListItem icon={<Branch />}>
            Tristique senectus et netus
          </EmptyState.ListItem>
          <EmptyState.ListItem icon={<Matrix />}>
            Mus mauris vitae ultricies
          </EmptyState.ListItem>
        </EmptyState.List>
        <EmptyState.Content>
          <Button>Button</Button>
        </EmptyState.Content>
      </EmptyState>,
    );

    expect(container).toMatchSnapshot();
  });

  it('renders snapshot in column direction and inverted', async () => {
    const { container } = render(
      <EmptyState
        title="Title Heading"
        image={<Rocket3d />}
        direction="column"
        inverted
      >
        <EmptyState.Content>Lorem ipsum dolor sit amet.</EmptyState.Content>
        <EmptyState.List>
          <EmptyState.ListItem icon={<Group />}>
            Auctor eu augue
          </EmptyState.ListItem>
          <EmptyState.ListItem icon={<Branch />}>
            Tristique senectus et netus
          </EmptyState.ListItem>
          <EmptyState.ListItem icon={<Matrix />}>
            Mus mauris vitae ultricies
          </EmptyState.ListItem>
        </EmptyState.List>
        <EmptyState.Content>
          <Button>Button</Button>
        </EmptyState.Content>
      </EmptyState>,
    );

    expect(container).toMatchSnapshot();
  });
});
