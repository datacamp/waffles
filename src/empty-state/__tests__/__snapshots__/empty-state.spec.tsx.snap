// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog renders snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  border-style: dashed;
  border-radius: 4px;
  border-width: 1px;
  padding: 24px;
  gap: 16px;
  height: 100%;
  width: 100%;
  border-color: rgba(5, 25, 45, 0.1);
  color: #05192D;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: left;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.emotion-0 ul {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.emotion-0 li {
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
}

.emotion-1 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 100px;
  height: 100px;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 16px;
  max-width: 650px;
}

.emotion-3 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  color: #05192D;
  font-weight: 800;
  font-size: 18px;
  margin-bottom: 8px;
  line-height: 1.25;
  line-height: 1.5;
  margin: 0;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0;
  padding: 16px 0;
  list-style: none;
  gap: 8px;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 16px;
  line-height: 1.25;
}

.emotion-5 svg {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-8 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  vertical-align: middle;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-weight: 800;
  line-height: 1;
  margin: 0;
  padding: 0;
  outline: 0;
  border-style: solid;
  border-radius: 4px;
  border-width: 2px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: background-color 125ms ease-out;
  transition: background-color 125ms ease-out;
  cursor: pointer;
  height: 36px;
  min-width: 36px;
  width: auto;
  padding-left: 16px;
  padding-right: 16px;
  font-size: 14px;
  color: #05192D;
  background-color: #03EF62;
  border-color: transparent;
}

.emotion-8::after {
  content: '';
  display: block;
  position: absolute;
  z-index: 1;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  margin: -4px;
  border-radius: 6px;
  pointer-events: none;
  -webkit-transition: box-shadow 125ms ease-out;
  transition: box-shadow 125ms ease-out;
}

.emotion-8:hover:not(:disabled) {
  background-color: #00D649;
}

.emotion-8:active:not(:disabled) {
  background-color: #03EF62;
}

.emotion-8:disabled {
  opacity: 0.6;
  cursor: default;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      Rocket3d
    </div>
    <div
      class="emotion-2"
    >
      <h2
        class="emotion-3"
      >
        Title Heading
      </h2>
      <div>
        Lorem ipsum dolor sit amet.
      </div>
      <ul
        class="emotion-4"
      >
        <li
          class="emotion-5"
        >
          Group
          Auctor eu augue
        </li>
        <li
          class="emotion-5"
        >
          Branch
          Tristique senectus et netus
        </li>
        <li
          class="emotion-5"
        >
          Matrix
          Mus mauris vitae ultricies
        </li>
      </ul>
      <div>
        <button
          class="emotion-8"
        >
          <span
            class="emotion-9"
          >
            Button
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Dialog renders snapshot in column direction 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  border-style: dashed;
  border-radius: 4px;
  border-width: 1px;
  padding: 24px;
  gap: 16px;
  height: 100%;
  width: 100%;
  border-color: rgba(5, 25, 45, 0.1);
  color: #05192D;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: left;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.emotion-0 ul {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.emotion-0 li {
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
}

.emotion-1 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 100px;
  height: 100px;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 16px;
  max-width: 650px;
}

.emotion-3 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  color: #05192D;
  font-weight: 800;
  font-size: 18px;
  margin-bottom: 8px;
  line-height: 1.25;
  line-height: 1.5;
  margin: 0;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0;
  padding: 16px 0;
  list-style: none;
  gap: 8px;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 16px;
  line-height: 1.25;
}

.emotion-5 svg {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-8 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  vertical-align: middle;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-weight: 800;
  line-height: 1;
  margin: 0;
  padding: 0;
  outline: 0;
  border-style: solid;
  border-radius: 4px;
  border-width: 2px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: background-color 125ms ease-out;
  transition: background-color 125ms ease-out;
  cursor: pointer;
  height: 36px;
  min-width: 36px;
  width: auto;
  padding-left: 16px;
  padding-right: 16px;
  font-size: 14px;
  color: #05192D;
  background-color: #03EF62;
  border-color: transparent;
}

.emotion-8::after {
  content: '';
  display: block;
  position: absolute;
  z-index: 1;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  margin: -4px;
  border-radius: 6px;
  pointer-events: none;
  -webkit-transition: box-shadow 125ms ease-out;
  transition: box-shadow 125ms ease-out;
}

.emotion-8:hover:not(:disabled) {
  background-color: #00D649;
}

.emotion-8:active:not(:disabled) {
  background-color: #03EF62;
}

.emotion-8:disabled {
  opacity: 0.6;
  cursor: default;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      Rocket3d
    </div>
    <div
      class="emotion-2"
    >
      <h2
        class="emotion-3"
      >
        Title Heading
      </h2>
      <div>
        Lorem ipsum dolor sit amet.
      </div>
      <ul
        class="emotion-4"
      >
        <li
          class="emotion-5"
        >
          Group
          Auctor eu augue
        </li>
        <li
          class="emotion-5"
        >
          Branch
          Tristique senectus et netus
        </li>
        <li
          class="emotion-5"
        >
          Matrix
          Mus mauris vitae ultricies
        </li>
      </ul>
      <div>
        <button
          class="emotion-8"
        >
          <span
            class="emotion-9"
          >
            Button
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Dialog renders snapshot in column direction and inverted 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  border-style: dashed;
  border-radius: 4px;
  border-width: 1px;
  padding: 24px;
  gap: 16px;
  height: 100%;
  width: 100%;
  border-color: rgba(255, 255, 255, 0.1);
  color: #FFFFFF;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: left;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.emotion-0 ul {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.emotion-0 li {
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
}

.emotion-1 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 100px;
  height: 100px;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 16px;
  max-width: 650px;
}

.emotion-3 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  color: #FFFFFF;
  font-weight: 800;
  font-size: 18px;
  margin-bottom: 8px;
  line-height: 1.25;
  line-height: 1.5;
  margin: 0;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0;
  padding: 16px 0;
  list-style: none;
  gap: 8px;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 16px;
  line-height: 1.25;
}

.emotion-5 svg {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-8 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  vertical-align: middle;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-weight: 800;
  line-height: 1;
  margin: 0;
  padding: 0;
  outline: 0;
  border-style: solid;
  border-radius: 4px;
  border-width: 2px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: background-color 125ms ease-out;
  transition: background-color 125ms ease-out;
  cursor: pointer;
  height: 36px;
  min-width: 36px;
  width: auto;
  padding-left: 16px;
  padding-right: 16px;
  font-size: 14px;
  color: #05192D;
  background-color: #03EF62;
  border-color: transparent;
}

.emotion-8::after {
  content: '';
  display: block;
  position: absolute;
  z-index: 1;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  margin: -4px;
  border-radius: 6px;
  pointer-events: none;
  -webkit-transition: box-shadow 125ms ease-out;
  transition: box-shadow 125ms ease-out;
}

.emotion-8:hover:not(:disabled) {
  background-color: #00D649;
}

.emotion-8:active:not(:disabled) {
  background-color: #03EF62;
}

.emotion-8:disabled {
  opacity: 0.6;
  cursor: default;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      Rocket3d
    </div>
    <div
      class="emotion-2"
    >
      <h2
        class="emotion-3"
      >
        Title Heading
      </h2>
      <div>
        Lorem ipsum dolor sit amet.
      </div>
      <ul
        class="emotion-4"
      >
        <li
          class="emotion-5"
        >
          Group
          Auctor eu augue
        </li>
        <li
          class="emotion-5"
        >
          Branch
          Tristique senectus et netus
        </li>
        <li
          class="emotion-5"
        >
          Matrix
          Mus mauris vitae ultricies
        </li>
      </ul>
      <div>
        <button
          class="emotion-8"
        >
          <span
            class="emotion-9"
          >
            Button
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Dialog renders snapshot in inverted 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  border-style: dashed;
  border-radius: 4px;
  border-width: 1px;
  padding: 24px;
  gap: 16px;
  height: 100%;
  width: 100%;
  border-color: rgba(255, 255, 255, 0.1);
  color: #FFFFFF;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: left;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.emotion-0 ul {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.emotion-0 li {
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
}

.emotion-1 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 100px;
  height: 100px;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 16px;
  max-width: 650px;
}

.emotion-3 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  color: #FFFFFF;
  font-weight: 800;
  font-size: 18px;
  margin-bottom: 8px;
  line-height: 1.25;
  line-height: 1.5;
  margin: 0;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0;
  padding: 16px 0;
  list-style: none;
  gap: 8px;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 16px;
  line-height: 1.25;
}

.emotion-5 svg {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-8 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  vertical-align: middle;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-weight: 800;
  line-height: 1;
  margin: 0;
  padding: 0;
  outline: 0;
  border-style: solid;
  border-radius: 4px;
  border-width: 2px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: background-color 125ms ease-out;
  transition: background-color 125ms ease-out;
  cursor: pointer;
  height: 36px;
  min-width: 36px;
  width: auto;
  padding-left: 16px;
  padding-right: 16px;
  font-size: 14px;
  color: #05192D;
  background-color: #03EF62;
  border-color: transparent;
}

.emotion-8::after {
  content: '';
  display: block;
  position: absolute;
  z-index: 1;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  margin: -4px;
  border-radius: 6px;
  pointer-events: none;
  -webkit-transition: box-shadow 125ms ease-out;
  transition: box-shadow 125ms ease-out;
}

.emotion-8:hover:not(:disabled) {
  background-color: #00D649;
}

.emotion-8:active:not(:disabled) {
  background-color: #03EF62;
}

.emotion-8:disabled {
  opacity: 0.6;
  cursor: default;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      Rocket3d
    </div>
    <div
      class="emotion-2"
    >
      <h2
        class="emotion-3"
      >
        Title Heading
      </h2>
      <div>
        Lorem ipsum dolor sit amet.
      </div>
      <ul
        class="emotion-4"
      >
        <li
          class="emotion-5"
        >
          Group
          Auctor eu augue
        </li>
        <li
          class="emotion-5"
        >
          Branch
          Tristique senectus et netus
        </li>
        <li
          class="emotion-5"
        >
          Matrix
          Mus mauris vitae ultricies
        </li>
      </ul>
      <div>
        <button
          class="emotion-8"
        >
          <span
            class="emotion-9"
          >
            Button
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
